app/controllers/nonsipp/landorpropertydisposal/IndividualBuyerNinoNumberController.scala

```diff
@@ -0,0 +1,122 @@
+  @Named("non-sipp") navigator: Navigator,
+  identifyAndRequireData: IdentifyAndRequireData,
+  formProvider: YesNoPageFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: ConditionalYesNoPageView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private val form: Form[Either[String, Nino]] = IndividualBuyerNinoNumberController.form(formProvider)
+
+  def onPageLoad(srn: Srn, landOrPropertyIndex: Max5000, disposalIndex: Max50, mode: Mode): Action[AnyContent] =
+    identifyAndRequireData(srn) { implicit request =>
+      val preparedForm =
+        request.userAnswers.fillForm(IndividualBuyerNinoNumberPage(srn, landOrPropertyIndex, disposalIndex), form)
+      Ok(view(preparedForm, viewModel(srn, landOrPropertyIndex, disposalIndex, mode)))
+
+    }
+
+  def onSubmit(srn: Srn, landOrPropertyIndex: Max5000, disposalIndex: Max50, mode: Mode): Action[AnyContent] =
+    identifyAndRequireData(srn).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors =>
+            Future
+              .successful(BadRequest(view(formWithErrors, viewModel(srn, landOrPropertyIndex, disposalIndex, mode)))),
+          value =>
+            for {
+              updatedAnswers <- Future
+                .fromTry(
+                  request.userAnswers
+                    .set(
+                      IndividualBuyerNinoNumberPage(srn, landOrPropertyIndex, disposalIndex),
+                      ConditionalYesNo(value)
+                    )
+                )
+              _ <- saveService.save(updatedAnswers)
+            } yield Redirect(
+              navigator
+                .nextPage(IndividualBuyerNinoNumberPage(srn, landOrPropertyIndex, disposalIndex), mode, updatedAnswers)
+            )
+        )
+    }
+}
+
+object IndividualBuyerNinoNumberController {
+  def form(formProvider: YesNoPageFormProvider): Form[Either[String, Nino]] = formProvider.conditional(
+    "individualBuyerNinoNumber.error.required",
+    mappingNo = Mappings.textArea(
+      "individualBuyerNinoNumber.no.conditional.error.required",
+      "individualBuyerNinoNumber.no.conditional.error.invalid",
+      "individualBuyerNinoNumber.no.conditional.error.length"
+    ),
+    mappingYes = Mappings.nino(
+      "individualBuyerNinoNumber.yes.conditional.error.required",
+      "individualBuyerNinoNumber.yes.conditional.error.invalid"
+    )
+  )
+
+  def viewModel(
+    srn: Srn,
+    landOrPropertyIndex: Max5000,
+    disposalIndex: Max50,
+    mode: Mode
+  ): FormPageViewModel[ConditionalYesNoPageViewModel] =
+    FormPageViewModel[ConditionalYesNoPageViewModel](
+      "individualBuyerNinoNumber.title",
+      Message("individualBuyerNinoNumber.heading", "John Doe"),
+      ConditionalYesNoPageViewModel(
+        yes = YesNoViewModel
+          .Conditional(Message("individualBuyerNinoNumber.yes.conditional", "John Doe"), FieldType.Input),
```

Comment:
I don't think this is right. Name should be the individual name that was typed in on the previous page.

Reply:
Hi @kberec , previous page is not yet merge in to main. I will fix all in one go. Just hard coded for instance.

Reply:
Looks it was just merged in.